fatiando/mesher.py

```diff
@@ -1720,3 +1720,433 @@ def keep(cell):
+    Create a triaxial ellipsoid.
+
+    .. note:: The coordinate system used is x -> North, y -> East
+    and z -> Down
+
+    Parameters:
+
+    * x, y, z : float
+        The coordinates of the center of the ellipsoid
+    * a, b, c : float
+        Semi-axes of the ellipsoid (a>b>c)
+    * alpha, gamma, delta
+        Orientation angles of the ellipsoid
+    * props : dict
+        Physical properties assigned to the ellipsoid.
+        Ex: ``props={'density':10,
+                     'remanence':[10000., 25., 40.],
+                     'k':[0.562, 0.485, 0.25, 90., 0., 0.]}``
+
+    Examples:
+
+        >>> e = TriaxialEllipsoid(1, 2, 3, 6, 5, 4, 10, 20, 30,{
+        ...                       'remanence':[10000., 25., 40.],
+        ...                       'k': [0.562, 0.485, 0.25, 90., 0., 0.]})
+        >>> e.props['remanence']
+        [10000.0, 25.0, 40.0]
+        >>> e.addprop('density', 20)
+        >>> print e.props['density']
+        20
+        >>> print e
+        x:1.0 | y:2.0 | z:3.0 | a:6.0 | b:5.0 | c:4.0 | alpha:10.0 | gamma:20.\
+0 | delta:30.0 | density:20 | k:[0.562, 0.485, 0.25, 90.0, 0.0, 0.0] | remanen\
+ce:[10000.0, 25.0, 40.0]
+        >>> e = TriaxialEllipsoid(1, 2, 3, 6, 5, 4, 10, 20, 30)
+        >>> print e
+        x:1.0 | y:2.0 | z:3.0 | a:6.0 | b:5.0 | c:4.0 | alpha:10.0 | gamma:20.\
+0 | delta:30.0
+        >>> e.addprop('density', 2670)
+        >>> print e
+        x:1.0 | y:2.0 | z:3.0 | a:6.0 | b:5.0 | c:4.0 | alpha:10.0 | gamma:20.\
+0 | delta:30.0 | density:2670
+
+    """
+
+    def __init__(self, x, y, z, a, b, c, alpha, gamma, delta, props=None):
+        GeometricElement.__init__(self, props)
+
+        self.x = float(x)
+        self.y = float(y)
+        self.z = float(z)
+        self.a = float(a)
+        self.b = float(b)
+        self.c = float(c)
+        self.alpha = float(alpha)
+        self.gamma = float(gamma)
+        self.delta = float(delta)
+
+        assert self.a > self.b and self.b > self.c, \
+            "a must be greater than b and b must greater than c"
+
+    def __str__(self):
+        """
+        Return a string representation of the ellipsoid.
+        """
+
+        names = [('x', self.x), ('y', self.y), ('z', self.z),
+                 ('a', self.a), ('b', self.b), ('c', self.c),
+                 ('alpha', self.alpha), ('gamma', self.gamma),
+                 ('delta', self.delta)]
+        names = names + [(p, self.props[p]) for p in sorted(self.props)]
+        return ' | '.join('%s:%s' % (n, v) for n, v in names)
+
+    def V(self, angles=None):
+        '''
+        Calculates the coordinate transformation matrix
+        for a triaxial model. The columns of this matrix
+        are defined according to the unit vectors v1, v2
+        and v3 presented by Clark et al. (1986, p. 192).
+
+        Parameters:
+
+        * angles: array like containing the three
+            orientation angles (alpha, gamma, delta) of the main
+            susceptibility axes.
```

Comment:
So it can depend on the susceptibility? Does it have to be? Couldn't this be just some other directional vector and then the magnetic modeling functions pass in the susceptibility? This is a matrix that converts from a given direction to the ellipsoid axis direction right?